shell/platform/embedder/embedder.h

```diff
@@ -1549,6 +1608,33 @@ FlutterEngineResult FlutterEngineSendPointerEvent(
     const FlutterPointerEvent* events,
     size_t events_count);
 
+//------------------------------------------------------------------------------
+/// @brief      Send a key event to the engine, causing the framework to
+///             dispatch a key event that is simply transformed from this one.
+///             The framework will decide whether to handle this event in a
+///             synchronous fashion, although due to technical limitation, the
+///             result is always reported asynchronously. The `callback` is
+///             guaranteed to be called exactly once.
+///
+/// @param[in]  engine         A running engine instance.
+/// @param[in]  event          The event data to be sent. This function will no
+///                            longer access `event` after returning.
+/// @param[in]  callback       The callback invoked by the engine when the
+///                            Flutter application has decided whether it handles
+///                            this event.
+/// @param[in]  user_data      The context associated with the callback. The exact
+///                            same value will used to invoke `callback`. Accepts
+///                            nullptr or a non-pointer value.
```

Comment:
Remove "or a non-pointer value"; it sounds like sending a pointer value is not allowed. Someone who knows what they are doing enough to pass data around as a `void *` is already familiar with that pattern and doesn't need to be told they can do it.